### DISCUSSION TITLE
make ui.open accept NiceGUI element as link target

### DISCUSSION THREAD

**Discussion by retsyo** [zauberzeug/nicegui#2105] [Ideas / Feature Requests]
[doc on Navigate on large pages](https://nicegui.io/documentation/link#navigate_on_large_pages) says we can simply pass a NiceGUI element as link target:
```python
label_B = ui.label('sth')
ui.link('Goto B', label_B)
```
which is a handy method. However, sometimes I don't need a `ui.link` but, for example, a `ui.button`. So I write
```python
def fnB():
    ui.open(label_B)

ui.button('Goto B', on_click=fnB)
```
however, I get
```
KeyError: <nicegui.elements.label.Label object at 0x000001D79AF728C0>
```

So for consistency in concept with `ui.link`, I suggest to make `ui.open` accept NiceGUI element as link target too.

p.s. for now, we can still use
```python
ui.link_target('target_A')
ui.label('sth')

def fnA():
    ui.open('#target_A')

ui.button('Goto A', on_click=fnA)
```

or
```python
with ui.button(''):
        ui.link('Goto B', label_B)
```

**Comment by falkoschindler** [comment]
Thanks for the suggestion, @retsyo! It's pretty convincing.
The implementation should be straight forward, given the code in `ui.link`. Would you like to create a pull request? 🙂
